docs/SNMP_Alarms.md

```diff
@@ -41,3 +41,19 @@ obtain the brief description, detailed description, and severity for the alarm:
 ## Alarm Models
 
 The alarm models used by Clearwater are defined in [alarmdefinition.h](https://github.com/Metaswitch/cpp-common/blob/master/include/alarmdefinition.h).
+
+## Alarm Resiliency
+
+Clearwater is designed so that if an EMS receiving alarm notifications fails and
+recovers, it will learn about any alarms raised by Clearwater while it was down,
+rather than those alarms just being lost. This is due to the following two
+features:
+
+*   The process responsible for sending SNMP notifications, the Alarm Agent,
+    keeps track of all the currently active error states in a table called the
+    Alarm Active Table. Upon restart, if supported, an EMS can read this table and
```

Comment:
I think we could tighten this up a bit more:

- I don't think saying "if supported" adds very much - it's ambiguous whether it;s Ckearwater or the EMS that might not support it, and the start of the next bullet ("If an EMS does not support reading the Active Alarm Table...") says the same thing more clearly
- "gain knowledge of" sounds a bit awkward to me - what about just saying "learn about"?